source/MaterialXCore/Document.cpp

```diff
@@ -157,12 +157,18 @@ void Document::initialize()
     setVersionString(DOCUMENT_VERSION_STRING);
 }
 
-void Document::importLibrary(ConstDocumentPtr library)
+void Document::importLibrary(ConstDocumentPtr library, const CopyOptions* copyOptions)
 {
+    bool skipDuplicateElements = copyOptions && copyOptions->skipDuplicateElements;
     for (ElementPtr child : library->getChildren())
     {
-        ElementPtr childCopy = addChildOfCategory(child->getCategory(), child->getName());
-        childCopy->copyContentFrom(child);
+        std::string childName = child->getName();
+        if (skipDuplicateElements && getChild(childName))
+        {
+            continue;
+        }
+        ElementPtr childCopy = addChildOfCategory(child->getCategory(), childName);
+        childCopy->copyContentFrom(child, copyOptions);
         if (!childCopy->hasSourceUri())
```

Comment:
Now that there's a sourceUris/copySourceUris flag passed to this method through CopyOptions, we should respect it here.